Webpack 5 generating wrong output in bundle.js using --env.production

### Issue Description

I try to upgrade our app from using @nativescript/webpack 4.1.0 to 5.0.0.

Let's assume the following snippet anywhere in the app code:

```
let test = {
    "myTestKey": "\uD835\uDCB6",      //ok
    "\uD835\uDCB6": "myTestValue"    //causing problem when running with --env.production
}
```

When running the app in development mode (tns run android|ios), everything works as expeced. The resulting bundle.js contains the following:
... { myTestKey: "𝒶", "𝒶" : "myTestValue" } ...



BUT when running this using **tns run android|ios --env.production**, the resulting bundle.js contains the following:
... { myTestKey: "𝒶", 𝒶: "myTestValue" } ...

So the second key of the object is not a string anymore, and now the app crashes with the following log:

> System.err: An uncaught Exception occurred on "main" thread.
> System.err: Unable to create application com.tns.NativeScriptApplication: com.tns.NativeScriptException: Cannot compile /data/data/org.nativescript.ns8wp5test/files/app/bundle.js
> System.err: SyntaxError: Invalid or unexpected token
> System.err: File: (file: app/webpack:/ns-8-wp-5-test/app/main-page.js:9:55)
> System.err: 
> System.err: StackTrace: 
> System.err: require(:1:266)
> System.err: 
> System.err: 
> System.err: SyntaxError: Invalid or unexpected token
> System.err: 
> System.err: StackTrace:
> System.err: java.lang.RuntimeException: Unable to create application com.tns.NativeScriptApplication: com.tns.NativeScriptException: Cannot compile /data/data/org.nativescript.ns8wp5test/files/app/bundle.js
> System.err: SyntaxError: Invalid or unexpected token
> System.err: File: (file: app/webpack:/ns-8-wp-5-test/app/main-page.js:9:55)
> System.err: 
> System.err: StackTrace: 
> System.err: require(:1:266)
> System.err: 
> System.err: 
> System.err: SyntaxError: Invalid or unexpected token
> System.err:     at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6465)
> System.err:     at android.app.ActivityThread.access$1300(ActivityThread.java:219)
> System.err:     at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1859)
> System.err:     at android.os.Handler.dispatchMessage(Handler.java:107)
> System.err:     at android.os.Looper.loop(Looper.java:214)
> System.err:     at android.app.ActivityThread.main(ActivityThread.java:7356)
> System.err:     at java.lang.reflect.Method.invoke(Native Method)
> System.err:     at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:492)
> System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:930)
> System.err: Caused by: com.tns.NativeScriptException: Cannot compile /data/data/org.nativescript.ns8wp5test/files/app/bundle.js
> System.err: SyntaxError: Invalid or unexpected token



In our real app, we use an [html entities decoder/encoder](https://github.com/mathiasbynens/he).

The 'encodeMap'object at [this line](https://github.com/mathiasbynens/he/blob/master/he.js#L32) contains some keys like shown in the code snippet above and this is causing the app to crash using webpack 5 in production mode.

As I said, it is working as expected with nativescript/webpack 5 in development mode (no minify/uglify) and when using nativescript/webpack 4.1.0 in general.


### Reproduction

Sample app: [ns-8-wp-5-test.zip](https://github.com/NativeScript/NativeScript/files/7728036/ns-8-wp-5-test.zip)

tns run android
or 
tns run ios
-> everything correct

tns run android --env.production
or
tns run ios --env.production
-> crash

### Relevant log output (if applicable)

_No response_

### Environment

Was not able to run the command for any reason

So here is the content content of package.json:

```
{
  "name": "ns-8-wp-5-test",
  "main": "app/app.js",
  "version": "1.0.0",
  "private": true,
  "dependencies": {
    "@nativescript/core": "^8.1.5"
  },
  "devDependencies": {
    "@nativescript/android": "8.1.1",
    "@nativescript/ios": "8.1.0",
    "@nativescript/webpack": "~5.0.0"
  }
}
```


The project contains the default webpack.config.js:

```
const webpack = require("@nativescript/webpack");

module.exports = (env) => {
	webpack.init(env);

	// Learn how to customize:
	// https://docs.nativescript.org/webpack

	return webpack.resolveConfig();
};

```

### Please accept these terms

- [X] I have searched the [existing issues](https://github.com/NativeScript/NativeScript/issues) as well as [StackOverflow](https://stackoverflow.com/questions/tagged/nativescript) and this has not been posted before
- [X] This is a bug report
- [X] I agree to follow this project's [Code of Conduct](https://github.com/NativeScript/NativeScript/blob/master/tools/notes/CONTRIBUTING.md#coc)

--- a/packages/webpack5/src/configuration/base.ts
+++ b/packages/webpack5/src/configuration/base.ts
@@ -157,6 +157,9 @@ export default function (config: Config, env: IWebpackEnv = _env): Config {
 				},
 				keep_fnames: true,
 				keep_classnames: true,
+				format: {
+					keep_quoted_props: true,
+				},
 			},
 		},
 	]);

